src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -654,13 +655,24 @@ private StateEditor doTraverse(State s0, RoutingRequest options, TraverseMode tr
      * @return penalty
      */
     private double calculatePenaltyFromParameters (String formula, float value, RequestParameters threshold, RequestParameters penalty){
-        Expression expression = new ExpressionBuilder(formula)
-                .variables("VALUE", "THRESHOLD", "PENALTY")
-                .build()
-                .setVariable("VALUE", value)
-                .setVariable("THRESHOLD", Double.parseDouble(threshold.getValue()))
-                .setVariable("PENALTY", Double.parseDouble(penalty.getValue()));
-        return expression.evaluate();
+        Map<String, Double> variables = new HashMap<>();
+        if (threshold != null && threshold.getValue() != null)
+            variables.put("THRESHOLD", Double.parseDouble(threshold.getValue()));
+        if (penalty != null && penalty.getValue() != null)
+            variables.put("PENALTY", Double.parseDouble(penalty.getValue()));
+        variables.put("VALUE", (double) value);
```

Comment:
```suggestion
        Map<String, Double> variables = new HashMap<>();
        
        if (threshold != null && threshold.getValue() != null)
            variables.put("THRESHOLD", Double.parseDouble(threshold.getValue()));
            
        if (penalty != null && penalty.getValue() != null)
            variables.put("PENALTY", Double.parseDouble(penalty.getValue()));
            
        variables.put("VALUE", (double) value);
```

Improves code readability